runtime/src/iree/base/loop_emscripten.c

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/base/loop_emscripten.h"
+
+#if defined(IREE_PLATFORM_EMSCRIPTEN)
+
+#include <emscripten.h>
+
+#include "iree/base/assert.h"
+
+//===----------------------------------------------------------------------===//
+// externs from loop_emscripten.js
+//===----------------------------------------------------------------------===//
+
+extern iree_status_t loop_command_call(iree_loop_callback_fn_t callback,
+                                       void* user_data, iree_loop_t loop);
+
+//===----------------------------------------------------------------------===//
+// iree_loop_emscripten_t
+//===----------------------------------------------------------------------===//
+
+typedef struct iree_loop_emscripten_t {
+  iree_allocator_t allocator;
+
+  // TODO(scotttodd): handle to a "scope"/object (managed in JS), so multiple
+  //                  loops can exist at once
+} iree_loop_emscripten_t;
+
+IREE_API_EXPORT iree_status_t iree_loop_emscripten_allocate(
+    iree_allocator_t allocator, iree_loop_emscripten_t** out_loop_emscripten) {
```

Comment:
nit: `loop` is fine in here vs `loop_emscripten` as there's no ambiguity (no `iree_loop_t loop;` used)